packages/protocol/scripts/truffle/deploy_release_contracts.ts

```diff
@@ -144,6 +147,92 @@ async function handleGrant(releaseGoldConfig: any, currGrant: number) {
   }
 }
 
+async function checkBalance(releaseGoldConfig: any) {
+  const weiAmountReleasedPerPeriod = new BigNumber(
+    web3.utils.toWei(releaseGoldConfig.amountReleasedPerPeriod.toString())
+  )
+  const grantTotal = weiAmountReleasedPerPeriod
+    .multipliedBy(releaseGoldConfig.numReleasePeriods)
+    .plus(startGold)
+    .toFixed()
+  while (true) {
+    const fromBalance = new BigNumber(await web3.eth.getBalance(fromAddress))
+    if (fromBalance.gte(grantTotal)) {
+      break
+    }
+    console.info(
+      chalk.yellow(
+        fromAddress +
+          "'s balance will not cover the next grant with identifier " +
+          releaseGoldConfig.identifier +
+          '.'
+      )
+    )
+    const addressResponse = await prompts({
+      type: 'text',
+      name: 'newFromAddress',
+      message:
+        'If you would like to continue deployment from a new address please provide it now. If you would like to cancel deployment just press enter. Grants already deployed will persist.',
+    })
+
+    if (addressResponse.newFromAddress === '') {
+      console.info('Exiting because of user input.')
+      process.exit(0)
+    }
+
+    if (!web3.utils.isAddress(addressResponse.newFromAddress)) {
+      console.info(
+        chalk.red('Provided address is invalid, please insert a valid Celo Address.\nRetrying.')
+      )
+      continue
+    }
+    console.info(
+      '\nSending 1 cGLD as a test from ' +
+        fromAddress +
+        ' to ' +
+        addressResponse.newFromAddress +
+        ' to verify ownership.\n'
+    )
+    await web3.eth.sendTransaction({
+      from: fromAddress,
+      to: addressResponse.newFromAddress,
+      value: web3.utils.toWei('1', 'ether'),
+    })
+    const confirmResponse = await prompts({
+      type: 'confirm',
+      name: 'confirmation',
+      message:
+        'Please check the balance of your provided address. You should see the 1cGLD transfer and an initial genesis balance if this is a shard from the genesis block.\nCan you confirm this (y/n)?',
+    })
+    if (!confirmResponse.confirmation) {
+      console.info(chalk.red('Setting new address failed.\nRetrying.'))
+      continue
+    }
+    console.info(chalk.green('\nTransfer confirmed, sending the remaining balance.\n'))
+    const fromBalancePostTransfer = new BigNumber(await web3.eth.getBalance(fromAddress))
+    await web3.eth.sendTransaction({
+      from: fromAddress,
+      to: addressResponse.newFromAddress,
+      value: fromBalancePostTransfer,
```

Comment:
Wait actually I'm surprised this works, we should not be able to transfer our entire balance since we need to pay for tx fees